pilot/pkg/config/kube/gateway/controller.go

```diff
@@ -95,11 +102,14 @@ func NewController(
 	var ctl *status.Controller
 
 	nsInformer := client.KubeInformer().Core().V1().Namespaces().Informer()
+	secretInformer := credentials.GetSecretInformer(client)
```

Comment:
I believe this will open a new watch to the api server; we don't want that. Note all the others like L104 are using the same shared informer factory, so this doesn't happen - if there is already another nsInformer L104 would just register a new handle for it. 

So we may need to find a way to share the underlying informer

Reply:
Possibly by not using the informer directly but instead the CredentialsController

Reply:
> Note all the others like L104 are using the same shared informer factory, so this doesn't happen - if there is already another nsInformer L104 would just register a new handle for it.

I think both L104 and L105 are creating an informer from the shared informer factory of the client?

L104:
```go
func (f *namespaceInformer) Informer() cache.SharedIndexInformer {
        // f.factory = client.KubeInformer()
	return f.factory.InformerFor(&corev1.Namespace{}, f.defaultInformer)
}
```

L105:
```go
func GetSecretInformer(client kube.Client) cache.SharedIndexInformer {
	return client.KubeInformer().InformerFor(&v1.Secret{}, func(k kubernetes.Interface, resync time.Duration) cache.SharedIndexInformer {
...
```
> I believe this will open a new watch to the api server

So given `CredentialsController` and `GatewayAPIController` share the same kube client, they should be sharing the same watch as well?

Reply:
I don't think so since the secret one uses custom config. Tou can check with --vklog=6 to istiod and look at the watch requests

Reply:
> I don't think so since the secret one uses custom config

The custom config has nothing to do with this, it is only used when creating a new informer. If an informer already exists, [`InformerFor` will just return the existing one
](https://github.com/kubernetes/client-go/blob/522eaa15ef6e0492b86105e1dd6ece67842bdf5f/informers/factory.go#L189-L197)

> Tou can check with --vklog=6 to istiod and look at the watch requests

Before: 
```bash
# kubectl logs istiod-55d9f784c7-l5zjr -n istio-system | grep "Listing and watching *v1.Secret" -F
2022-11-23T02:25:39.621661Z     info    klog    Listing and watching *v1.Secret from k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167
2022-11-23T02:25:39.636595Z     info    klog    Listing and watching *v1.Secret from k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167
```

After:
```bash
# kubectl logs istiod-767c8d8445-vw4k8 -n istio-system | grep "Listing and watching *v1.Secret" -F
2022-11-23T03:23:56.606211Z     info    klog    Listing and watching *v1.Secret from pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169
2022-11-23T03:23:56.629189Z     info    klog    Listing and watching *v1.Secret from pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169
```

It's expected that both set up 2 watches, one for [credentials](https://github.com/istio/istio/blob/1.15.3/pilot/pkg/credentials/kube/secrets.go#L79), the other for [multi cluster secrets](https://github.com/istio/istio/blob/1.15.3/pkg/kube/multicluster/secretcontroller.go#L117)

Reply:
thanks, I didn't know that.